fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_weapons.dm

```diff
@@ -0,0 +1,481 @@
+#define upgraded_val(x,y) ( CEILING((x * (1.07 ** y)), 1) )
+#define CALIBER_BLOODSILVER "bloodsilver"
+#define WEAPON_UPGRADE "weapon_upgrade"
+
+/obj/item/melee/trick_weapon
+	icon = 'fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/icons/weapons.dmi'
+	lefthand_file = 'fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/icons/weapons_lefthand.dmi'
+	righthand_file = 'fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/icons/weapons_righthand.dmi'
+	///upgrade level of the weapon
+	var/upgrade_level = 0
+	///base force when transformed
+	var/on_force
+	///base force when in default state
+	var/base_force
+	///default name of the weapon
+	var/base_name
+	///is the weapon in its transformed state?
+	var/enabled = FALSE
+
+
+/obj/item/melee/trick_weapon/proc/upgrade_weapon()
+	SIGNAL_HANDLER
+
+	upgrade_level++
+	force = upgraded_val(base_force,upgrade_level)
+	var/datum/component/transforming/transform = src.GetComponent(/datum/component/transforming)
+	transform.force_on = upgraded_val(on_force,upgrade_level)
+
+
+/obj/item/melee/trick_weapon/attack(mob/target, mob/living/user, params) //our weapon does 25% less damage on non monsters
+	var/old_force = force
+	if(!(target.mind?.has_antag_datum(/datum/antagonist/changeling)) && !IS_BLOODSUCKER(target) && !IS_HERETIC(target))
+		force = force * 0.75
+	..()
+	force = old_force
+
+/obj/item/melee/trick_weapon/darkmoon
+	name = "Darkmoon Greatsword"
+	base_name = "Darkmoon Greatsword"
+	desc = "Ahh my guiding moonlight, you were by my side all along."
+	icon_state = "darkmoon"
+	inhand_icon_state = "darkmoon_hilt"
+	w_class = WEIGHT_CLASS_SMALL
+	slot_flags = ITEM_SLOT_BELT
+	block_chance = 20
+	on_force = 20
+	base_force = 17
+	throwforce = 12
+	damtype = BURN
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	attack_verb_continuous = list("attacks", "slashes", "stabs", "slices", "tears", "lacerates", "rips", "dices", "cuts")
+	attack_verb_simple = list("attack", "slash", "stab", "slice", "tear", "lacerate", "rip", "dice", "cut")
+	///ready to launch a beam attack?
+
+	COOLDOWN_DECLARE(moonbeam_fire)
+
+
+/obj/item/melee/trick_weapon/darkmoon/Initialize(mapload)
+	. = ..()
+	force = base_force
+	AddComponent(/datum/component/transforming, \
```

Comment:
Why not add a tranforming trick weapon subtype to avoid this copy paste?

Reply:
not all the trick weapons use this component (axe)